neuro_san/internals/run_context/langchain/llms/standard_langchain_llm_factory.py

```diff
@@ -81,29 +88,38 @@ def create_base_chat_model(self, config: Dict[str, Any]) -> BaseLanguageModel:
             ChatOpenAI = resolver.resolve_class_in_module("ChatOpenAI",
                                                           module_name="langchain_openai.chat_models.base",
                                                           install_if_missing="langchain-openai")
+
+            # Our run-time model resource here is httpx client which we need to control directly:
+            openai_proxy = self.get_value_or_env(config, "openai_organization", "OPENAI_PROXY")
+            request_timeout = config.get("request_timeout")
+            http_client = httpx.AsyncClient(proxy=openai_proxy, timeout=request_timeout)
+
+            # Now build an OpenAI client using this resource:
+            async_openai_client: AsyncOpenAI = AsyncOpenAI(
+                api_key=self.get_value_or_env(config, "openai_api_key",
+                                              "OPENAI_API_KEY"),
+                base_url=self.get_value_or_env(config, "openai_api_base", "OPENAI_API_BASE"),
+                organization=self.get_value_or_env(config, "openai_organization", "OPENAI_ORG_ID"),
+                timeout=request_timeout,
+                max_retries=config.get("max_retries"),
+                http_client=http_client
+            )
+
```

Comment:
Note that all constructor parameters of ChatOpenAI are now distributed between http_client,
AsyncOpenAI and ChatOpenAI. According of course to GPT recommendations.